src/main/java/org/jnode/fs/xfs/Superblock.java

```diff
@@ -62,11 +62,19 @@ public int getBlockSize() {
         return (int) getUInt32(0x4);
     }
 
+    public long getTotalBlocks() throws IOException {
+        return (int) getInt64(0x8);
```

Comment:
The cast to int in this function and the next is not necessary.

Reply:
code change applied